### ISSUE TITLE
NASM Runner does use any code in Preloaded section

### ISSUE THREAD

**Issue by nomennescio** [Codewars/codewars-runner-cli#619]
When trying to create a NASM translation, I noticed that none of the code in the Preloaded section is used by the Runner. Even if I type in complete rubbish, it is totally ignored, no compile or link error whatsoever.

**Comment by kazk** [comment]
Yeah, I didn't add it because I wasn't sure what to support (NASM or C) and wanted to keep it simple at first. Codewars UI should hide the preloaded editor though.

**Comment by nomennescio** [comment]
In my opinion, preloaded should contain NASM code because you usually hide your solution in there.

**Comment by kazk** [comment]
For random tests? For NASM, the reference solution can be written in C for that purpose.

**Comment by nomennescio** [comment]
Yes for random tests. If you use NASM there, you can just reuse your own solution for it.
If you want, you can still code another solution in C under the 'Test Cases'.

Having preloaded in the same language as the solution language is as far as I can see standard practice for all languages.

**Comment by error256** [comment]
There's little point to make it in C because there's already a module in C, the tests. The question is whether the code in Preloaded should be assembled separately or concatenated with user code. In the latter case it could contain some compile-time things like macros.
